frontend/src/common/queries/collections.ts

```diff
@@ -281,44 +277,45 @@ async function collectionUploadLinks({
   return json.dataset_uuid;
 }
 
-export function useCollectionUploadLinks(
-  id: string,
-  visibility: VISIBILITY_TYPE
-) {
+export function useCollectionUploadLinks(id: string) {
   const queryCache = useQueryClient();
 
   return useMutation(collectionUploadLinks, {
     onSuccess: () => {
-      queryCache.invalidateQueries([USE_COLLECTION, id, visibility]);
+      queryCache.invalidateQueries([USE_COLLECTION, id]);
     },
   });
 }
 
 async function deleteCollection({
   collectionID,
-  visibility = VISIBILITY_TYPE.PRIVATE,
 }: {
   collectionID: Collection["id"];
-  visibility?: VISIBILITY_TYPE;
 }) {
-  const baseUrl = apiTemplateToUrl(API_URL + API.COLLECTION, {
+  const finalURL = apiTemplateToUrl(API_URL + API.COLLECTION, {
     id: collectionID,
   });
-  const finalUrl = baseUrl + `?visibility=${visibility}`;
 
-  const response = await fetch(finalUrl, DELETE_FETCH_OPTIONS);
+  const response = await fetch(finalURL, DELETE_FETCH_OPTIONS);
 
   if (!response.ok) {
     throw await response.json();
   }
 }
 
 export function useDeleteCollection(
-  id = "",
-  visibility = VISIBILITY_TYPE.PRIVATE
-) {
+  collection: Collection | TombstonedCollection | null | undefined
+): UseMutationResult<
+  void,
+  unknown,
+  { collectionID: string },
+  { previousCollections: CollectionResponsesMap }
+> {
```

Comment:
@tihuan Is this appropriate typing here?

Reply:
discussion: Wondering if the interface could be the same primitive types, so the arguments are more predictable, and we can just set default `""` directly there? 

https://www.johndcook.com/blog/2011/07/19/you-wanted-banana/

The return type looks good!

E.g.,

```ts
function useDeleteCollection(id = "", visibility = ""): UseMutationResult<...>  { ... } 
```

Reply:
agreed, was trying out something new, but I think it makes sense this way

Reply:
Note though, this introduced _a lot_ of prop drilling:
https://github.com/chanzuckerberg/single-cell-data-portal/pull/2197/commits/450bcb143c2e324a87798a21ed22408b4ce270f1

Reply:
Woah 7 levels lol The deepest I've seen so far ahaha

Maybe time to pass `id` and `visibility` as context?